src/main/java/qna/service/QnAService.java

```diff
@@ -33,26 +30,18 @@ public Question findQuestionById(Long id) {
     }
 
     @Transactional
-    public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteException {
+    public void deleteQuestion(User loginUser, Long questionId) throws CannotDeleteException {
         Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
+
+        question.checkAuthority(loginUser);
+        question.isHasAnswersIsNotOwner(loginUser);
+
+        DeleteHistories deleteHistories = new DeleteHistories();
         question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
+
+        deleteHistories.add(question);
+        deleteHistories.add(question.getAnswers());
+
         deleteHistoryService.saveAll(deleteHistories);
```

Comment:
대부분의 로직을 도메인에 위임해보시는건 어떠실까요? 

아래와 같은 형태로 리팩토링 해보시면 어떠실지 의견드립니다.!  
```suggestion
        Question question = findQuestionById(questionId);
        deleteHistoryService.saveAll(question.delete(loginUser));
```

Reply:
의견 감사드립니다! 말씀하신대로 수정하면 훨씬 깔끔하게 코드가 정리되네요! 한가지 질문 드리고 싶은데요.
question 도메인에서 deleteHistories 객체가 생성되어 return 되어도 괜찮을지?  deleteHistories 객체가 question에 의존적인 형태가 아닐지 아니면 단순 로그 저장용이기 때문에 괜찮은건지, 이 부분에 대한 의견이 궁금합니다! 감사합니다

Reply:
넵! question에 의존적인 형태는 아니라고 봅니다! :)